http/handler_store.go

```diff
@@ -153,6 +153,41 @@ func (s *storeHandler) GetCollection(rw http.ResponseWriter, req *http.Request)
 	}
 }
 
+func (s *storeHandler) GetSchema(rw http.ResponseWriter, req *http.Request) {
+	store := req.Context().Value(storeContextKey).(client.Store)
+
+	switch {
```

Comment:
suggestion: This switch should follow the same order as the one in the cli. Although we are constructing the request in the client to match exactly one of these cases, there is nothing preventing someone from manually writing a request with multiple parameters.

Reply:
The client interface does not support that, so this does not support that.  Which means the CLI should not support that.  Which means I was wrong to ask Keenan to make the collection describe CLI stuff to support that.

Reply:
This means that in both cases (CLI and HTTP for both schema and collection) we should make the params mutually exclusive.

Reply:
Yeah I think so :(

Reply:
Although, I'd see changing the collection stuff as out of scope here, and with the code freeze coming up a rework could me that users end up getting nothing for 8 more weeks.

I think I'm in favour of leaving both Schema and Collection funcs as is in this PR, and then changing them both together in another ticket.

Reply:
We could do that. It would simplify the interfaces but add a bit of complexity to the entry point. Not a big deal. I assume passing `nil` would return all schemas?

Reply:
> I assume passing nil would return all schemas?

If we do this I would want to use `immutable.Option` for everything, types would be:

```
struct Filter {
  Name Option[String]
  Root Option[String]
  ID Option[String]
}
interface {
  GetSchemas(Option[Filter])
}
```

Reply:
I think `Option[Filter]` is overkill and would be annoying more than anything. I would say it's overkill for all of this but I mind it less for the fields than for the function parameter.

Reply:
> I think `Option[Filter]` is overkill and would be annoying more than anything. I would say it's overkill for all of this but I mind it less for the fields than for the function parameter.

Could have:
```
interface {
  GetSchemas(...Filter)
}
```
Maybe, but assuming you were happy enough with doing this in another ticket it is not important to sort it out here.

Reply:
Totally ok with doing this in a different ticket. In the mean time do you mind just ordering the switch statement the same as the CLI so that they behave the same way?

Reply:
> ordering the switch statement the same as the CLI

Yes of course, sorry I forgot about that.

- [ ] Change switch statement
- [ ] Open ticket for Filter stuff

Reply:
https://github.com/sourcenetwork/defradb/issues/2008